rfcs/20190411-tensorboard-improved-plugin-ext.md

```diff
@@ -0,0 +1,151 @@
+# TensorBoard: Improved Plugin Extensibility
+
+| Status        | Proposed       |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Stephan Lee (Google), William Chargin (Google) |
+| **Sponsor**   | Mani Varadarajan (Google)                 |
+| **Updated**   | 2019-04-11                                           |
+
+## Overview
+
+This RFC proposes a friction-free approach to enabling TensorBoard contributors to create and share TensorBoard plugins as installable extensions.
+
+TensorBoard is composed of various plugins that show up as separate dashboards to users (e.g., scalars, graphs, hparams, histograms, etc.). Today. anyone can contribute a new plugin by forking TensorBoard, adding their plugin, and getting the PR merged. However, this can require a lengthy review to make sure it is architected correctly and is aligned with TensorBoard’s goals.
+
+This RFC covers two changes that will make this process easier:
+1. Plugins can be distributed as Python packages. Users can install them as extensions to TensorBoard.
+2. Plugins are no longer limited to use Polymer but can instead use any frontend library or framework (e.g., React)
+
+The intention is to enable any prospective contributor to easily be able to create or reuse a visualization with tools familiar to them, and enable users to try it out quickly. Plugins that are useful to many users should still be added to TensorBoard, but this removes the friction needed to start getting feedback or trying out something new.
```

Comment:
Might be good to clarify "added to TensorBoard" means "added to the codebase of TensorBoard and the tensorboard package directly" in this case, to avoid confusion with "added to TensorBoard via the 3p plugin route".

Reply:
@GalOshri did we mean that it should be part of the "first-party plugin" if it is good?

Reply:
Yes, @caisq's clarification is useful. I think we'd need to think through these details, but the idea is that if some functionality is super useful to most users, it should be a part of the default experience.